lib/codecs/src/encoding/format/native_json.rs

```diff
@@ -98,7 +98,12 @@ mod tests {
         serializer
             .encode(histogram_event.clone(), &mut bytes)
             .unwrap();
-        let json = serializer.to_json_value(histogram_event).unwrap();
-        assert_eq!(bytes.freeze(), serde_json::to_string(&json).unwrap());
+        // TODO: remove this comment. Add a PR comment stating that converting the histogram_event to a serde_json::Value
+        // Changed the order of the keys to alphabetical order and the string comparison failed due to the Event struct
+        // not serializing the keys in the same order as the serde_json::Value.
+        assert_eq!(
```

Comment:
This test was broken for me in the master branch, I don't know if I should remove this change, tell me.

The issue was that `serializer.to_json_value(histogram_event)` returned a `serde_json::Value`, which when serialized with `serde_json::to_string(&json)`, its keys were alphabetically sorted.

When serializing the `histogram_event` with `serializer
            .encode(histogram_event.clone(), &mut bytes)`, the fields are not alphabetically sorted, serde does not enforce this in arbitrary struct serialization (it does on `Value` struct serializations).

I think the test should be change as stated in this diff. Please, tell me your opinions about this

Reply:
Huh, interesting. It does pass in CI but you are correct that it may be relying on some non-deterministic behavior. The change seems fine to me.